app/controllers/admin/users_controller.rb

```diff
@@ -6,6 +6,22 @@ def new
       @user = User.new
     end
 
+    def create
+      # there is no usual user creation process when ICHAIN auth is enabled
+      if ENV['OSEM_ICHAIN_ENABLED'] != 'true'
+        @user = User.new(user_params)
+        @user.skip_confirmation!
+        if @user.save
+          redirect_to admin_users_path, notice: 'User successfully created.'
+        else
+          flash.now[:error] = "Creating User failed: #{@user.errors.full_messages.join('. ')}."
+          render :new
+        end
+      else
+        redirect_to admin_users_path, error: 'Cannot create user with ICHAIN auth enabled.'
```

Comment:
Cancancan takes care of this with https://github.com/commandprompt/osem-pg/blob/f9a816eed44625683746192c43acd8fc53f92999/app/controllers/admin/users_controller.rb#L3
So this if statement should not be necessary here, as we should only be able to access the action `create`, if the abilities allow that.

If the authorization fails, cancancan raises an error https://github.com/openSUSE/osem/blob/master/app/controllers/application_controller.rb#L46